templates/terraform/sweeper_file.go.erb

```diff
@@ -97,12 +98,25 @@ func testSweep<%= sweeper_name -%>(region string) error {
 	nonPrefixCount := 0
 	for _, ri := range rl {
 		obj := ri.(map[string]interface{})
+		<% if delete_id -%>
+		var name string
+		// Id detected in the delete URL, attempt to use id.
+		if obj["id"] != nil {
+			name = GetResourceNameFromSelfLink(obj["id"].(string))
```

Comment:
I chose to write these changes in a way that will only cause diffs for a minority of sweepers. Reusing the `name` variable to hold the `id` instead of just renaming the variable allows me to do that. But if it's important enough for the variable name to reflect that it _could_ be an id, I can make that change.